app/services/jobs/return-logs/generate-return-logs.service.js

```diff
@@ -13,16 +13,30 @@ const {
   cycleStartDate,
   formatDateObjectToISO
 } = require('../../../lib/dates.lib.js')
+const FetchReturnCycleService = require('./fetch-return-cycle.service.js')
+const GenerateReturnCycleService = require('./generate-return-cycle.service.js')
 
 /**
  * Generates the payload for submission to the returns table.
  *
  * @param {Array} returnRequirements - the return requirements to be turned into return logs
- * @param {string} returnCycleId - the UUID of the return cycle
  *
  * @returns {Promise<Array>} the array of return log payloads to be created in the database
  */
-async function go (returnRequirements, returnCycleId) {
+async function go (returnRequirements) {
+  const today = formatDateObjectToISO(new Date())
+
+  let allYearReturnCycleId = await FetchReturnCycleService.go(today, false)
+  let summerReturnCycleId = await FetchReturnCycleService.go(today, true)
+
+  if (!allYearReturnCycleId) {
+    allYearReturnCycleId = await GenerateReturnCycleService.go(false)
+  }
+
+  if (!summerReturnCycleId) {
+    summerReturnCycleId = await GenerateReturnCycleService.go(true)
+  }
```

Comment:
This has added a lot to the existing function and is distinctly different to what it was doing before. Therefore, I'd suggest moving this to a private function that is responsible for returning the return cycle IDs needed (whether they be fetched or generated). This then becomes a single line call in the `go()`.